efficientdet/efficientdet_arch.py

```diff
@@ -154,46 +131,28 @@ def _maybe_apply_1x1(feat):
         feat = _maybe_apply_1x1(feat)
       height_stride_size = int((height - 1) // target_height + 1)
       width_stride_size = int((width - 1) // target_width + 1)
-      if pooling_type == 'max' or pooling_type is None:
-        # Use max pooling in default.
-        feat = tf.layers.max_pooling2d(
-            inputs=feat,
-            pool_size=[height_stride_size + 1, width_stride_size + 1],
-            strides=[height_stride_size, width_stride_size],
-            padding='SAME',
-            data_format=data_format)
-      elif pooling_type == 'avg':
-        feat = tf.layers.average_pooling2d(
-            inputs=feat,
-            pool_size=[height_stride_size + 1, width_stride_size + 1],
-            strides=[height_stride_size, width_stride_size],
-            padding='SAME',
-            data_format=data_format)
-      else:
-        raise ValueError('Unknown pooling type: {}'.format(pooling_type))
+
+      # Use max pooling in default.
+      feat = tf.layers.max_pooling2d(
+          inputs=feat,
+          pool_size=[height_stride_size + 1, width_stride_size + 1],
+          strides=[height_stride_size, width_stride_size],
+          padding='SAME',
+          data_format=data_format)
+
       if conv_after_downsample:
         feat = _maybe_apply_1x1(feat)
     elif height <= target_height and width <= target_width:
       feat = _maybe_apply_1x1(feat)
       if height < target_height or width < target_width:
-        height_scale = target_height // height
-        width_scale = target_width // width
-        if (use_native_resize_op or target_height % height != 0 or
-            target_width % width != 0):
-          if data_format == 'channels_first':
-            feat = tf.transpose(feat, [0, 2, 3, 1])
-          feat = tf.cast(
-              tf.image.resize_nearest_neighbor(
-                  tf.cast(feat, tf.float32), [target_height, target_width]),
-              dtype=feat.dtype)
-          if data_format == 'channels_first':
-            feat = tf.transpose(feat, [0, 3, 1, 2])
-        else:
-          feat = nearest_upsampling(
-              feat,
-              height_scale=height_scale,
-              width_scale=width_scale,
-              data_format=data_format)
+        if data_format == 'channels_first':
+          feat = tf.transpose(feat, [0, 2, 3, 1])
+        feat = tf.cast(
+            tf.image.resize_nearest_neighbor(
```

Comment:
I deleted the customized resize_nearest_neighbor (it was for some legacy issues, not used in the current code).